#! /bin/sh -e
# sparc64-build.dpatch by  <schizo@debian.org>
#
# DP: Allows sparc64 to build more simply

if [ $# -eq 3 ] && [ "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
	patch -f --no-backup-if-mismatch -p0 < $0;;
    -unpatch)
	patch -f --no-backup-if-mismatch -R -p0 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1;;
esac

exit 0

diff -urNad gcc/config/sparc/linux64.h sparc64-build.gcc-3.3/src/gcc/config/sparc/linux64.h
--- gcc/config/sparc/linux64.h	2003-03-13 08:40:33.000000000 +0000
+++ gcc/config/sparc/linux64.h	2003-03-13 08:46:18.000000000 +0000
@@ -37,8 +37,8 @@
    + MASK_STACK_BIAS + MASK_APP_REGS + MASK_FPU + MASK_LONG_DOUBLE_128)
 #endif
 
-#undef ASM_CPU_DEFAULT_SPEC
-#define ASM_CPU_DEFAULT_SPEC "-Av9a"
+#undef ASM_CPU64_DEFAULT_SPEC
+#define ASM_CPU64_DEFAULT_SPEC "-Av9a"
 
 #ifdef SPARC_BI_ARCH
 
--- gcc/config.gcc	2004-06-29 01:44:04.000000000 -0400
+++ gcc/config.gcc	2004-07-05 10:46:52.813558531 -0400
@@ -1828,8 +1828,17 @@
 	use_fixproto=yes
 	;;
 sparc-*-linux*)		# SPARC's running GNU/Linux, libc6
+	# If cpu is specified, assume we want a 32/64 compiler
+	if test x$with_cpu = x; then
 	tm_file="${tm_file} dbxelf.h elfos.h svr4.h sparc/sysv4.h sparc/linux.h"
 	tmake_file="${tmake_file} sparc/t-crtfm"
+	else
+	tm_file="sparc/biarch64.h ${tm_file} dbxelf.h elfos.h svr4.h sparc/sysv4.h sparc/linux64.h"
+	tmake_file="${tmake_file} sparc/t-linux64 sparc/t-crtfm"
+	float_format=sparc
+	fi
+#	tm_file="${tm_file} dbxelf.h elfos.h svr4.h sparc/sysv4.h sparc/linux.h"
+#	tmake_file="${tmake_file} sparc/t-crtfm"
 	;;
 sparc-*-rtems*)
 	tm_file="${tm_file} dbxelf.h elfos.h svr4.h sparc/sysv4.h sol2.h sparc/sol2.h sparc/elf.h sparc/rtemself.h rtems.h"
